### ISSUE TITLE
Add a MultiProvider implementation for metrics/multi package

### ISSUE THREAD

**Issue by medzin** [go-kit/kit#593]
Currently there is no implementation of a [Provider](https://github.com/go-kit/kit/blob/master/metrics/provider/provider.go) interface for the [metrics/multi](https://godoc.org/github.com/go-kit/kit/metrics/multi) package. Adding a `MultiProvider` that accepts a slice of other `Provider` implementations would make using `multi` package more convenient.

**Comment by peterbourgon** [comment]
Metrics are being refactored for the next release; although there is no ETA, this work is probably not worth doing...

**Comment by medzin** [comment]
Will the new version of the metrics package support the use of multiple providers for one metric? It is very convenient to use a combination of expvar and some external metrics storage.

**Comment by peterbourgon** [comment]
Yes. The current version of the metrics package also does this just fine, for the record; just not with the Provider interface.

```
c := multi.NewCounter(
    prometheus.NewCounter(...),
    expvar.NewCounter(...),
)
```
